packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx

```diff
@@ -197,24 +197,22 @@ export const DiagramRenderer = memo(({ diagramRefreshedEventPayload }: DiagramRe
       ) {
         setNodes((oldNodes) => applyNodeChanges(noReadOnlyChanges, oldNodes));
       } else {
-        setNodes((oldNodes) => {
-          resetHelperLines(changes);
-          let transformedNodeChanges: NodeChange[] = transformBorderNodeChanges(noReadOnlyChanges, oldNodes);
-          transformedNodeChanges = transformUndraggableListNodeChanges(transformedNodeChanges);
-          transformedNodeChanges = transformResizeListNodeChanges(transformedNodeChanges);
-          transformedNodeChanges = applyHelperLines(transformedNodeChanges);
+        resetHelperLines(changes);
+        let transformedNodeChanges: NodeChange[] = transformBorderNodeChanges(noReadOnlyChanges, nodes);
+        transformedNodeChanges = transformUndraggableListNodeChanges(transformedNodeChanges);
+        transformedNodeChanges = transformResizeListNodeChanges(transformedNodeChanges);
+        transformedNodeChanges = applyHelperLines(transformedNodeChanges);
 
-          let newNodes = applyNodeChanges(transformedNodeChanges, oldNodes);
+        let newNodes = applyNodeChanges(transformedNodeChanges, nodes);
 
-          newNodes = applyMoveChange(transformedNodeChanges, newNodes);
-          newNodes = applyHandleChange(transformedNodeChanges, newNodes as Node<NodeData, DiagramNodeType>[]);
+        newNodes = applyMoveChange(transformedNodeChanges, newNodes);
+        newNodes = applyHandleChange(transformedNodeChanges, newNodes as Node<NodeData, DiagramNodeType>[]);
 
-          layoutOnBoundsChange(transformedNodeChanges, newNodes as Node<NodeData, DiagramNodeType>[]);
-          return newNodes;
-        });
+        layoutOnBoundsChange(transformedNodeChanges, newNodes as Node<NodeData, DiagramNodeType>[]);
+        setNodes(newNodes);
       }
     },
-    [setNodes, layoutOnBoundsChange, getNodes, getEdges]
+    [layoutOnBoundsChange, getNodes, getEdges]
```

Comment:
We're still using `setNodes` inside the callback; is it voluntary to have removed the dependency?

Reply:
Yes, setState function are guaranteed to have a stable identity so we can omit it from useCallBack and useEffect